File: NRDB/BrowserCell.m
//
//  BrowserCell.m
//  NRDB
//
//  Created by Gereon Steffens on 08.08.14.
//  Copyright (c) 2014 Gereon Steffens. All rights reserved.
//


#import "BrowserCell.h"
#import "BrowserResultViewController.h"

@implementation BrowserCell

-(void) moreClicked:(UIButton*)sender
{
    [BrowserResultViewController showPopupForCard:self.card inView:self fromRect:sender.frame];
}


@end

File: NetDeck/BrowserCell.swift
//
//  BrowserCell.swift
//  NetDeck
//
//  Created by Gereon Steffens on 31.12.16.
//  Copyright © 2019 Gereon Steffens. All rights reserved.
//

import UIKit

protocol PopupDisplayer: class {
    func showPopup(for card: Card, in view: UIView, from rect: CGRect)
}

class BrowserCell: UITableViewCell {
    @IBOutlet weak var moreButton: UIButton!
    @IBOutlet weak var nameLabel: UILabel!
    
    weak var parent: PopupDisplayer?
    
    var card: Card! {
        didSet {
            self.setCard(card)
        }
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()
        self.selectionStyle = .none
    }
    
    @IBAction func moreClicked(_ sender: UIButton) {
        parent?.showPopup(for: self.card, in: self, from: sender.frame)
    }
    
    func setCard(_ card: Card) {
        fatalError("must be overridden")
    }
}
